pkg/v1/tkg/test/tkgctl/shared/e2e_common.go

```diff
@@ -223,3 +227,110 @@ func E2ECommonSpec(ctx context.Context, inputGetter func() E2ECommonSpecInput) {
 		By("Test successful !")
 	})
 }
+
+func GetNextAvailableTkrVersion(tkgctlClient tkgctl.TKGClient, currentTkrVersion string) (string, error) {
+	var foundVersion string
+
+	tkrs, err := tkgctlClient.GetTanzuKubernetesReleases("")
+	if err != nil {
+		return "", err
+	}
+
+	for i := range tkrs {
+		if !isCompatible(tkrs[i]) {
+			continue
+		}
+
+		if _, exists := tkrs[i].Labels[runv1alpha3.LabelDeactivated]; exists {
+			continue
+		}
+
+		specVersionIsNewer, err := isNewerVMwareVersion(tkrs[i].Spec.Version, currentTkrVersion)
```

Comment:
if the spec version (from the tkr we are looking at) is newer  (and compatible) we compare it to any version we already found,  to pick the older of the two. This will ensure we pick the next avaiable version: (newer than our currentVersion, but the older tan all other newer )